### PR TITLE
Upgrades to node & base images for node and python

### PR DESCRIPTION
**Pull request by davidwilby**
Maintenance has ended for v14 and v18 is the current LTS version of node.

### CODE REVIEW

**Comment by ianroberts** [comment]
We'll also need to update the `Dockerfile` to use the 18 base image in `nodebuilder` and install the right version of node in the `test` stage.

File: Dockerfile
Comment (ianroberts) [inline Dockerfile:1]: While we're at it with upgrading dependencies, I suggest we switch from `buster` to `bullseye` for the node and python base images - bullseye is the more recent Debian release, and in particular I've noticed that the backend image is currently based on `python:3.9-slim-buster` but the `apt-get` command that runs during the build installs the Debian packages for python 3.7 in there as well, adding bloat if nothing else.  This doesn't happen with the bullseye base image.

Comment (davidwilby) [reply Dockerfile:1]: Have done in 1404d57 - but where's the `apt-get` call that installs the python 3.7 packages? In the buster base image?

Comment (ianroberts) [reply Dockerfile:1]: They're coming in at the `apt-get -y install gcc libpq-dev libmagic1 postgresql-client` stage, presumably as transitive dependencies of something else.

**Pull request merged**